pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/ExtensibleLoadManagerImpl.java

```diff
@@ -477,7 +477,7 @@ public CompletableFuture<Optional<String>> selectAsync(ServiceUnitId bundle,
                                                            Set<String> excludeBrokerSet) {
         BrokerRegistry brokerRegistry = getBrokerRegistry();
         return brokerRegistry.getAvailableBrokerLookupDataAsync()
-                .thenCompose(availableBrokers -> {
+                .thenComposeAsync(availableBrokers -> {
```

Comment:
@lifepuzzlefun I think we can switch a thread here.

Reply:
agree